### ISSUE TITLE
revertFunc is not provided for eventDrop and eventResize

### ISSUE THREAD

**Issue by vz28bh** [Jamaks/ng-fullcalendar#71]
FullCalendar provides a revertFunc which can be used to undo a user input but I can't find any reference to that function in the event.detail object. This would be used for example if the user tried to drag an event to an invalid day and you want to force the event back to where it came from.

**Comment by Jamaks** [comment]
Hi! 
revertFunc not supported, I have no idea how to do it, maybe in the future I will add a new `@Input` `[(revertFunc)]`

**Comment by TyreX6** [comment]
You can make it working by doing this in the updateEvent:
`model = {
      event: {
        id: model.event.id,
        id_res: model.event.id_res,
        start: model.event.start,
        end: model.event.end,
        title: model.event.title,
        className: model.event.className,
        editable: model.event.editable,
        // other params
      },
      duration: model.duration
    };
this.events[model.event.id].start = moment(model.event.start).subtract(model.duration, 'seconds');
this.events[model.event.id].end = moment(model.event.end).subtract(model.duration, 'seconds');
`

Of course you need to import:
`import * as jquery from 'jquery';
import * as moment from 'moment';`

**Comment by sean-liang** [comment]
I also figured a workaround which is reset all events:

@ViewChild(Calendar)
ucCalendar: Calendar;

revert() {
  this.ucCalendar.fullCalendar("removeEvents");
  this.ucCalendar.fullCalendar("renderEvents", this.events);
}

**Comment by Jamaks** [comment]
I close this issue, becouse @sean-liang seems to have found a solution

**Comment by dolf-o-matric** [comment]
IMO not really a solution, because Sean-liang's code snippet reverts ALL the events, not just the currently dragged/resized-one! It would be nice to have a revertFunc, just like in the original FullCalendar's resize/drag callback event payload to support some runtime validation features.

**Comment by cheddrsuite** [comment]
If you change:
        this.options.eventDrop = (event, duration) => {
            let /** @type {?} */ detail = { event: event, duration: duration };

to:
        this.options.eventDrop = (event, duration, revertFunc) => {
            let /** @type {?} */ detail = { event: event, duration: duration, revertFunc: revertFunc };
You will get the revert function along with the event.

**Comment by anakin59490** [comment]
@cheddrsuite 
You're right but when the function is called, the event still stay on new date :
It's strange because I see the event goes back to initial date then moves to the new date..

How did you use it ?
Thx
